### ISSUE TITLE
tiff-test.sh not running on OSX

### ISSUE THREAD

**Issue by piponazo** [Exiv2/exiv2#190]
We recently noticed that the system tests for TIFF are not running under the OSX travis jobs:

```
Running tiff-test.sh ...
./tiff-test.sh requires bash v4 or greater.  Running 3.2.57(1)-release.  ./tiff-test.sh skipped.
```

Investigate if we can upgrade the bash version in the OSX images or make those tiff-tests work without that requirement.

**Comment by clanmills** [comment]
MacOS-X ships with bash 3.  Apparently, Apple cannot upgrade to 4 due to licensing issues.

I believe you said that we can install _more or less_ anything on the travis node.  I have bash 4 on my laptop (in /usr/local/bin/bash).  It only requires support from 'standard' libraries:

```
550 rmills@rmillsmbp:~/clanmills $ otool -L /usr/local/bin/bash
/usr/local/bin/bash:
	/usr/lib/libncurses.5.4.dylib (compatibility version 5.4.0, current version 5.4.0)
	/usr/local/lib/libintl.8.dylib (compatibility version 10.0.0, current version 10.2.0)
	/usr/lib/libiconv.2.dylib (compatibility version 7.0.0, current version 7.0.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1197.1.1)
551 rmills@rmillsmbp:~/clanmills $ ls -alt $(which bash)
-rwxr-xr-x  1 rmills  admin  1199192 27 Jul  2014 /usr/local/bin/bash
552 rmills@rmillsmbp:~/clanmills $ 
```
Does travis always run on the same 'node'?  We could create a directory contrib/macosx/bin (or something like that) to hold it.

Another (and probably better suggestion) is to rewrite the test in C++ and put it in the samples directory (or bash3 or python).  That test was originally based on an executable called tiff-probe that was only available on Linux.  I rewrote it as a bash-script and later discovered that MacOS-X doesn't ship bash4.  Incidentally, Apple now protect /bin/bash (and many other utilities) from being changed by the user.

If you rewrite it in C++, you'll learn more about the structure of tiff files.

So .... Mr P, what's your judgement about the best remedy?

**Comment by D4N** [comment]
I'd suggest that we integrate these tests into the new python test suite too. It will require more work than the regression test suite, but in the end we are only comparing the output of exiv2 to an expected output (albeit with a lot of preprocessing in between, but that can be done in python too).

**Comment by piponazo** [comment]
hi guys. Good points. I think we can wait for the python test suite and take care of this situation there. Anyways this issue has been present in travis-ci since the beginning. We just noticed about it recently. 

Should I close the issue or you think it is good to leave it open as a remainder ?

**Comment by D4N** [comment]
I'd leave it open for now.

**Comment by clanmills** [comment]
I think we're all in enthusiastic agreement.  The "all new python test suite" will resolve this.  Let's close this issue as it's totally understood and we have an action plan to address this capability.  However the "tiff-test.sh" will go away, so this issue should be closed.

Who's going to write "pythonic tiff-test?".  Clearly there's a useful learning opportunity here to learn about the structure of tiff and the python test suite.  I'm willing to take care of this, however I'm happy to encourage you to take on this challenge!

**Comment by D4N** [comment]
Fixed by #155.
